vowpalwabbit/io/logger.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+
+#include <memory>
+#include <utility>
+#include <spdlog/spdlog.h>
+
+namespace VW
+{
+namespace io
+{
+  /*
+  // TODO: should this be an object to be passed around or stand-alone 
+struct logger {
+private:
+  std::shared_ptr<spdlog::logger> _internal_logger;
+public:
+  vw_logger()
+  : _internal_logger(spdlog::default_logger()) {}
+  
+  template<typename FormatString, typename... Args>
+    void log_info(const FormatString &fmt, Args&&...args)
+  {
+    _internal_logger->info(fmt, std::forward<Args>(args)...);
+  }
+
+  template<typename FormatString, typename... Args>
+    void log_error(const FormatString &fmt, Args&&...args)
+  {
+    _internal_logger->error(fmt, std::forward<Args>(args)...);
+  }
+
+
+  template<typename FormatString, typename... Args>
+    void log_critical(const FormatString &fmt, Args&&...args)
+  {
+    _internal_logger->critical(fmt, std::forward<Args>(args)...);
+  }
+
+};
+  */
+
+namespace logger
+{
+  enum class log_level { trace, debug, info, warn, error, critical, off };
```

Comment:
If we only support a subset of these we should remove the ones we dont use

Reply:
Are we going to support trace and debug? It seems as though they may be too verbose and add too many runtime log checks?

Reply:
I'm not sure, it depends it can be useful for debugging purposes, but I agree it adds a lot of branching statements. We could always use a compile-time flag to no-op some of the more verbose logging, but I'm not really a fan of having 2 ways to turn off logging

Reply:
The multiple ways of turning off logging already exists...